src/components/auth/login.js

```diff
@@ -0,0 +1,176 @@
+          errors.user_name =
+        value.length < 5
+        ? 'must be a valid user name'
+        : '';
+        break;
+        case 'password':
+          errors.password =
+          value.length < 4
+          ? 'Password must be more than 4 characters long'
+          : '';
+          break;
+          default:
+            break;
+      }
+  
+      this.setState({errors, [name]: value}, ()=> {
+        
+      })
+  };
+  
+ 
+  
+  loginUser = e => {
+    e.preventDefault();
+    //console.log(user)
+
+      axiosWithAuth()
+      .post("/login", this.state.user)
+      .then(res => {
+
+        localStorage.setItem('token', res.data.token);
+        //redirect to Groas dashboard page
+        if (this.user.user_name && this.user.password) {
+          this.props.history.push('/');
+        }
+    
+      })
+      .catch(err => console.log(err));
+      
+  };
+  
+
+  render() {
+    const {errors} = this.state;
+  return (
+   
+    <div className='LoginPage'>
+      {/* Container - ENTIRE PAGE */}
+      <div className='container'>
+        <div className='bartop'>
+           
+              {/* <img className='Logo' src={Groa} alt='GroaLogo'/> */}
+              <NavLink to='/register'><h2 className='Register'>Register</h2></NavLink>    
+        </div>
+        
+        <div className='Middle'>
+          <div className='boxLeft'>
+          
+          <img className='Logo2' src={Groa2} alt='GroaLogo'/>
+           
+          </div>
+{/* END BOX LEFT */}
+        <div className='boxRight'>
+          <form className='LoginForm' onSubmit={this.loginUser}>
+          <h1 className='textlogin'>Log in</h1>
+        
+          <div className={'a' + (this.submitted && !this.user.user_name ? 'error' : '')}/>
+          <input className='input1'  
+              type="text"
+              name="user_name"
+              value={this.user_name}
+              onChange={this.handleChange}
+              placeholder="Username or email required"
+              errorMessage='username required'
+            />
+{/* ERROR MESSAGE */}
+      {errors.user_name.length > 0 && 
+        <span className='error'>{errors.user_name}</span>}
+
+            <input className='input1'
+              type="password"
+              name="password"
+              value={this.password}
+              onChange={this.handleChange}
+              placeholder="Password"
+            />
+
+      {errors.password.length > 0 && 
+        <span className='error'>{errors.password}</span>}
+            
+            <div className='TextandCheck'>
+                    <div className='CheckBoxContainer'>
+                        <input className='Checkbox' type="checkbox"/>
+                    </div>
+                    <h5> Remember me</h5>
+                    <div className='TextandCheck2'>
+                      <h5>Forgot password?</h5>
+                    </div>
+            </div>
+            
+            <i class="fa fa-search searchIcon"></i>
```

Comment:
I'm not sure what this search icon is for, and I cannot visually see it in the staged url.